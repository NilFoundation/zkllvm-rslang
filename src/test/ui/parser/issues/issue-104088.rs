fn test() {
    if let 123 = 123 { println!("yes"); }
}

fn test_2() {
    let 1x = 123;
    //~^ ERROR expected identifier, found number literal
}

fn test_3() {
    let 2x: i32 = 123;
    //~^ ERROR expected identifier, found number literal
}

fn test_4() {
    if let 2e1 = 123 {
        //~^ ERROR mismatched types
    }
}

fn test_5() {
    let 23name = 123;
    //~^ ERROR expected identifier, found number literal
}

fn main() {}
