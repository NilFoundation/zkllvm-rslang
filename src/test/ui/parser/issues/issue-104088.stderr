error: expected identifier, found number literal
  --> $DIR/issue-104088.rs:6:9
   |
LL |     let 1x = 123;
   |         ^^ identifiers cannot start with a number

error: expected identifier, found number literal
  --> $DIR/issue-104088.rs:11:9
   |
LL |     let 2x: i32 = 123;
   |         ^^ identifiers cannot start with a number

error: expected identifier, found number literal
  --> $DIR/issue-104088.rs:22:9
   |
LL |     let 23name = 123;
   |         ^^^^^^ identifiers cannot start with a number

error[E0308]: mismatched types
  --> $DIR/issue-104088.rs:16:12
   |
LL |     if let 2e1 = 123 {
   |            ^^^   --- this expression has type `{integer}`
   |            |
   |            expected integer, found floating-point number

error: aborting due to 4 previous errors

For more information about this error, try `rustc --explain E0308`.
